### ISSUE TITLE
Change the save behaviour of .twinproj file to include validation before replace

### ISSUE THREAD

**Issue by WaynePhillipsEA** [WaynePhillipsEA/twinbasic#259]
**Is your feature request related to a problem? Please describe.**
In response to #258, I suggest the save behaviour for twinproj files be tweaked with the aim of preventing a bad save of a twinproj file.

**Describe the solution you'd like**
The twinproj file should initially be saved with a different extension, e.g. .twinproj.partial and then only after the file has been validated as being OK delete the current .twinproj file and rename the .partial file to the final .twinproj extension.

**Additional context**
There should be some extra sanity checks in the validation process to ensure the file doesn't get saved in a bad state (e.g. an empty file...), in case the compiler or VS code process gets badly screwed up before the save occurs.

**Comment by bclothier** [comment]
Agreed but I have one belt-and-suspender suggestion; those operations should be using transactions (if not already) so that you have a clean way for rollbacking no matter what the specific error was.

**Comment by WaynePhillipsEA** [comment]
File I/O is actually handled by Node.js, and as far as I'm aware it doesn't offer transactional file operations.  Save new -> verify -> delete old -> rename approach should be pretty robust.
